examples/timeseries/timeseries_transformer_classification.py

```diff
@@ -0,0 +1,189 @@
+"""
+
+import tensorflow.keras as keras
+import tensorflow.keras.backend as K
+
+"""
+Here, we make the MultiHeadAttention layer to perform
+self-attention, by generating each of {query, key, value}
+matrices, through a dense layer.
+This corresponds to implementing an attention mechanism by:
+`Q = dense_q(x), K = dense_k(x), V = dense_v(x)` , where
+each `dense_x()` is a different instance of `keras.layers.Dense`.
+
+(The actual implementation is a much more optimized version.)
+"""
+
+class SelfAttention(keras.Model):
+    def __init__(self, head_size, num_heads=1, dropout=0, name='SelfAttention', **kwargs):
+        super().__init__(name=name, **kwargs)
+        self.attention = keras.layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=head_size, dropout=dropout, **kwargs)
+
+    # input shape = output shape = (batch size, seq len, features)
+    def call(self, inputs, **kwargs):
+        x = inputs
+        return self.attention(query=x, key=x, value=x, **kwargs)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+"""
+The Self-Attention mechanism is weak on it's own. We need to add a residual connection, (layer) normalization,
+dropout and projection layers for this to work. The following layer can also be stacked multiple times.
+
+The projection layers are implemented through `keras.layers.Conv1D`, by (automatically) broadcasting the compute to the last feature dimension of the input tensors.
+"""
+
+class SelfAttentionBlock(keras.Model):
+    def __init__(self,head_size, num_heads=1, ff_dim=None, dropout=0, name='SelfAttentionBlock',  **kwargs):
+        super().__init__(name=name, **kwargs)
+
+        if ff_dim is None:
+            ff_dim = head_size
+
+        self.attention = SelfAttention(head_size, num_heads, dropout=dropout)
+        self.attention_dropout = keras.layers.Dropout(dropout)
+        self.attention_norm = keras.layers.LayerNormalization(epsilon=1e-6)
+
+        self.ff_conv1 = keras.layers.Conv1D(
+            filters=ff_dim, kernel_size=1, activation='relu')
+
+        # self.ff_conv2 at build()
+        self.ff_dropout = keras.layers.Dropout(dropout)
+        self.ff_norm = keras.layers.LayerNormalization(epsilon=1e-6)
+
+    def build(self, input_shape):
+        self.ff_conv2 = keras.layers.Conv1D(
+            filters=input_shape[-1], kernel_size=1)
+
+    def call(self, inputs, training, **kwargs):
+        x = self.attention_norm(inputs, **kwargs)
+        x = self.attention(x, **kwargs)
+        x = self.attention_dropout(x, training=training, **kwargs)
+        res = x + inputs
+
+        x = self.ff_norm(res, **kwargs)
+        x = self.ff_conv1(x, **kwargs)
+        x = self.ff_dropout(x, training=training, **kwargs)
+        x = self.ff_conv2(x, **kwargs)
+        x = self.ff_dropout(x, training=training, **kwargs)
```

Comment:
Basically this entire 5 lines block can be a call to a Sequential. Basically you can replace the whole thing with

```python
class TransformerEncoder(layers.Layer):
  def __init__(self, head_size, num_heads, ff_dim, dropout=0, **kwargs):
      super().__init__(**kwargs)
      self.attention = layers.MultiHeadAttention(
          key_dim=head_size, num_heads=num_heads, dropout=dropout)
      self.attention_dropout = layers.Dropout(dropout)
      self.attention_norm = layers.LayerNormalization(epsilon=1e-6)
  
  def build(self, input_shape):
      self.ff_net = keras.Sequential([
          layers.LayerNormalization(epsilon=1e-6),
          layers.Conv1D(filters=ff_dim, kernel_size=1, activation='relu'),
          layers.Dropout(dropout),
          layers.Conv1D(filters=input_shape[-1], kernel_size=1),
      ])
  
  def call(self, inputs, training=False):
      x = self.attention_norm(inputs)
      x = self.attention(x, x, training=training)
      x = self.attention_dropout(x, training=training)
      res = x + inputs
      x = self.ff_net(res, training=training)
      return x + res
```